src/share/streams_nuopc/dshr_strdata_mod.F90

```diff
@@ -1266,6 +1271,16 @@ subroutine shr_strdata_readstrm(sdat, ns, stream, stream_mesh, &
           rcode = pio_get_att(pioid, varid, "_FillValue", fillvalue_r4)
        else if (pio_iovartype == PIO_DOUBLE) then
           rcode = pio_get_att(pioid, varid, "_FillValue", fillvalue_r8)
+       else if (pio_iovartype == PIO_SHORT) then
```

Comment:
I think that there could be short data type without add_offset and scale_factor such as land-sea mask etc. So, i think the interface just need to check those attributes and use them if they are available otherwise the data can be passed directly from PIO.

Reply:
I think that I would like to wait until we encounter such a case before we deal with it - is that okay with you?

Reply:
Okay, that is fine. Thanks for the short data type support!